processing/src/main/java/io/druid/segment/BitmapOffset.java

```diff
@@ -33,21 +33,69 @@
   private final IntIterator itr;
   private final BitmapFactory bitmapFactory;
   private final ImmutableBitmap bitmapIndex;
+  private final boolean descending;
 
   private volatile int val;
 
-  public BitmapOffset(BitmapFactory bitmapFactory, ImmutableBitmap bitmapIndex)
+  public BitmapOffset(BitmapFactory bitmapFactory, ImmutableBitmap bitmapIndex, boolean descending)
   {
     this.bitmapFactory = bitmapFactory;
     this.bitmapIndex = bitmapIndex;
-    this.itr = bitmapIndex.iterator();
+    this.descending = descending;
+    this.itr = newIterator();
     increment();
   }
 
+  private IntIterator newIterator()
+  {
+    if (!descending) {
+      return bitmapIndex.iterator();
+    }
+    // ImmutableRoaringReverseIntIterator is not cloneable.. looks like a bug
+    // update : it's fixed in 0.5.13
+    int i = bitmapIndex.size();
+    int[] back = new int[bitmapIndex.size()];
```

Comment:
With roaring bitmap-0.5.13, we can iterate in inverse order (0.5.4 currently). And other bitmaps do not support that. In some time I'll make issue for addressing that.